main.tf

```diff
@@ -70,13 +70,30 @@ module "efs" {
   vpc_id             = "${var.vpc_id}"
   subnets            = "${var.private_subnets}"
   availability_zones = "${var.availability_zones}"
-  security_groups    = ["${module.elastic_beanstalk_environment.security_group_id}"]             # EB/EC2 instances are allowed to connect to the EFS
   zone_id            = "${var.zone_id}"
+
+  # EB/EC2 instances and DataPipeline instances are allowed to connect to the EFS
+  security_groups = ["${module.elastic_beanstalk_environment.security_group_id}", "${module.efs_backup.security_group_id}"]
+}
+
+# EFS backup to S3
+module "efs_backup" {
+  source                             = "git::https://github.com/cloudposse/terraform-aws-efs-backup.git?ref=tags/0.3.2"
+  name                               = "${var.name}"
+  stage                              = "${var.stage}"
+  namespace                          = "${var.namespace}"
+  region                             = "${var.aws_region}"
+  vpc_id                             = "${var.vpc_id}"
+  efs_mount_target_id                = "${element(module.efs.mount_target_ids, 0)}"
+  use_ip_address                     = "true"
```

Comment:
It seems to be not necessary unless we're using custom DNS setting in `DHCP Options Set`. When DNS name of EFS is being prevented from resolving.

Reply:
Why do we need to have the `use_ip_address` attribute and use IP addresses at all?
`EFS` generates the private DNS names for the `EFS` and for each mount points.
AWS does not like IP addresses.

Reply:
In one of our customer's infrastructure, their DNS system is "broken" for usage inside AWS because it does not work with the split-horizon DNS. This was a workaround for that.